tg_bot/modules/locks.py

```diff
@@ -208,6 +283,17 @@ def del_lockables(bot: Bot, update: Update):
                         chat.kick_member(new_mem.id)
                         message.reply_text("Only admins are allowed to add bots to this chat! Get outta here.")
             else:
+                #allow whitelisted URLs
+                if lockable == 'url':
+                    entities = set(message.parse_entities([MessageEntity.URL]).values())
+
+                    # MessageEntity.TEXT_LINK could be added in the filter above, but would return the text, not the
+                    # url. So I must add all entities that have a 'url' field separately
+                    entities = entities | set(entity.url for entity in message.entities if entity.url)
+                    #if all URLs are any of the whitelisted ones, accept the message
+                    if all( any(regexp.search(text) for regexp in sql.get_whitelist(chat.id).values())
```

Comment:
How do you handle invalid regex expressions? if any are broken, this will blow up and entirely break whitelisting for that chat. I would personally not allow regex for this, as most users have very limited knowledge of it.

Reply:
Users don't get to interact with any of this as regex. All they do is send URLs. Only text that Telegram considers a URL is added (via entities). Also, the URLs themselves are escaped with `re.escape` before the regex is compiled.